staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -68,6 +69,68 @@ type action struct {
 	AllNamespaces bool // true iff the action is namespaced but works on aggregate result for all namespaces
 }
 
+func ConvertGroupVersionIntoToDiscovery(list []metav1.APIResource) []apidiscoveryv2beta1.APIResourceDiscovery {
```

Comment:
Given the KEP and statement, this may return an error (if a discovery doc reports something impossible, like two subresources on the same parent defining different scopes cluster/not) but shouldn't for anything that isn't impossible.  If it returns an error, then conversion to the new type is impossible (negotiation shouldn't allow it) AND it should possibly generate an error message in the apiserver startup as well as potentially display a warning to the client during discovery (if you ask for aggregated, and aggregated was impossible, return non-aggregated and warn, but don't warn if you ask for non-aggregated)

Reply:
See the previous comment, I think those cases should come from aggregation with a non-golang/extremely outdated apiserver. This construction portion in a golang apiserver should have those cases as impossible (unless we're trying to relax this?)

Reply:
If this is something we can detect very early, I would go with a panic. If this is something that is extremely unlikely to happen, I think I'd be fine with an error log (that's probably my least favorite option since these things tend to just be ignored). I'd say either verify at start-up and panic, or return an  error?